### PR TITLE
Parse Boolean types by default

### PR DESCRIPTION
**Pull request by austinweisgrau**
Commit 766cfaedc5652af8c39fde890067a99b6fa58518 created a feature for parsing boolean types but turned it off by default. This commit turns that feature on by default and adds a comment about how to turn it off and what that does.

When this feature is turned on, parsons will detect boolean column types when copying a parsons Table to a database and create a boolean column in the database. The default behavior without this change is to coerce booleans to be strings when loading to database.

Partially addresses #942

### CODE REVIEW

**Comment by austinweisgrau** [comment]
This will change default behavior for loading booleans from parsons Table to a database, so implementations based on the prior default behavior will likely break. e.g.

```
tbl = Table([{"bool_col": True}])
Redshift.copy(tbl, "tablename")
Redshift().query("select * from tablename where bool_col = 'True'")
```
will no longer work, instead the more normal code will be needed:
```
Redshift().query("select * from tablename where bool_col ")
```

**Comment by IanRFerguson** [comment]
> Ok so digging deeper, it looks like the current implementation will accept any of the following case-insensitive python values as True: `[True, "true", "t", 1, "1", "yes"]` and any of the following as False: `[False, "false", "f", 0, "0", "no"]`
> 
> I think this is bad and the only values we should translate into boolean are python booleans, `[True, False]`
> 
> If someone wants a set of strings or integers to be coerced into booleans when they're loading a parsons Table to a database, they should do that type conversion explicitly themself. It seems sketchy to have strings like "t" or "yEs" getting coerced into booleans by default.
> 
> These changes are made with [1cc6bd1](https://github.com/move-coop/parsons/commit/1cc6bd1e44d855cc9853f954666ffad189db9ff4)
> 
> What do you think?

I agree philosophically, we shouldn't be coercing data types implicitly so this is a valuable change. Going to take one more look at the diff but overall looks good

File: parsons/databases/database/constants.py [hunk 1]
```diff
@@ -153,11 +153,9 @@
 
 VARCHAR = "varchar"
 FLOAT = "float"
-
-DO_PARSE_BOOLS = False
 BOOL = "bool"
-TRUE_VALS = ("TRUE", "T", "YES", "Y", "1", 1)
```
Comment (IanRFerguson) [inline parsons/databases/database/constants.py:159]: wowza

File: parsons/databases/database/database.py [hunk 2]
```diff
@@ -161,64 +138,45 @@ def detect_data_type(self, value, cmp_type=None):
         # Stop if the compare type is already a varchar
         # varchar is the highest data type.
         if cmp_type == self.VARCHAR:
```
Comment (IanRFerguson) [inline parsons/databases/database/database.py:140]: So much cleaner

File: parsons/databases/database/constants.py (hunk 1 above)
Comment (austinweisgrau) [reply parsons/databases/database/constants.py:159]: Oh yeah I should delete those lines lol, they're not used anymore

**Review by Jason94 — APPROVED** [review]
This looks really great. Thanks @austinweisgrau for getting this over the finish line.

**Pull request merged**